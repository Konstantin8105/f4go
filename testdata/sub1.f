       subroutine square_cube(i,isquare,icube)
         integer, intent(in)  :: i             ! input
         integer, intent(out) :: isquare,icube ! output
         isquare = i**2
         icube   = i**3
       end subroutine square_cube
    
       program xx
           implicit none
           integer :: i,isq,icub
           i = 4
           call square_cube(i,isq,icub)
           print*,"i,i^2,i^3=",i,isq,icub
       end program xx
