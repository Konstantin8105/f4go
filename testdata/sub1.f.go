package main

func square_cube() {
	{
		{
			var tempVarF4GO_32 int
			var tempVarF4GO_45 int
			tempVarF4GO_45 = *i
			tempVarF4GO_32 = tempVarF4GO_45 * tempVarF4GO_45
			*isquare = tempVarF4GO_32
		}
		{
			var tempVarF4GO_34 int
			var tempVarF4GO_49 int
			tempVarF4GO_49 = *i
			var tempVarF4GO_51 int
			tempVarF4GO_51 = tempVarF4GO_49 * tempVarF4GO_49
			tempVarF4GO_34 = tempVarF4GO_49 * tempVarF4GO_51
			*icube = tempVarF4GO_34
		}
	}
}
func xx() {
	{
		var i int
		i = 4
		square_cube(i, isq, icub)
		{
			var dt_parm.0 __st_parameter_dt
			dt_parm.0.common.filename = "testdata/sub1.f "[1]
			dt_parm.0.common.line = 13
			dt_parm.0.common.flags = 128
			dt_parm.0.common.unit = 6
			_gfortran_st_write(dt_parm.0)
			_gfortran_transfer_character_write(dt_parm.0, "i,i^2,i^3="[1], 10)
			_gfortran_transfer_integer_write(dt_parm.0, i, 4)
			_gfortran_transfer_integer_write(dt_parm.0, isq, 4)
			_gfortran_transfer_integer_write(dt_parm.0, icub, 4)
			_gfortran_st_write_done(dt_parm.0)
		}
		{
			var dt_parm.1 __st_parameter_dt
			dt_parm.1.common.filename = "testdata/sub1.f "[1]
			dt_parm.1.common.line = 14
			dt_parm.1.common.flags = 128
			dt_parm.1.common.unit = 6
			_gfortran_st_write(dt_parm.1)
			_gfortran_transfer_character_write(dt_parm.1, "i,i^2,i^3="[1], 10)
			_gfortran_transfer_integer_write(dt_parm.1, i, 4)
			_gfortran_transfer_integer_write(dt_parm.1, isq, 4)
			_gfortran_transfer_integer_write(dt_parm.1, icub, 4)
			_gfortran_st_write_done(dt_parm.1)
		}
	}
}
func main() int {
	{
		var options.2 [288]int
		_gfortran_set_args(argc, argv)
		_gfortran_set_options(9, options.2[0])
		xx()
		return f4goUndefinedmodify_expr
	}
}
