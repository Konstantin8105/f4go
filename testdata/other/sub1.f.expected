SUBROUTINE          	subroutine
IDENT               	square_cube
(                   	(
IDENT               	i
,                   	,
IDENT               	isquare
,                   	,
IDENT               	icube
)                   	)
NEW_LINE            	

INTEGER             	integer
,                   	,
IDENT               	intent
(                   	(
IDENT               	in
)                   	)
DOUBLE_COLON        	::
IDENT               	i
COMMENT             	! input
NEW_LINE            	

INTEGER             	integer
,                   	,
IDENT               	intent
(                   	(
IDENT               	out
)                   	)
DOUBLE_COLON        	::
IDENT               	isquare
,                   	,
IDENT               	icube
COMMENT             	! output
NEW_LINE            	

IDENT               	isquare
=                   	=
IDENT               	i
                    	**
INT                 	2
NEW_LINE            	

IDENT               	icube
=                   	=
IDENT               	i
                    	**
INT                 	3
NEW_LINE            	

END                 	end
SUBROUTINE          	subroutine
IDENT               	square_cube
NEW_LINE            	

NEW_LINE            	

PROGRAM             	program
IDENT               	xx
NEW_LINE            	

IMPLICIT            	implicit
IDENT               	none
NEW_LINE            	

INTEGER             	integer
DOUBLE_COLON        	::
IDENT               	i
,                   	,
IDENT               	isq
,                   	,
IDENT               	icub
NEW_LINE            	

IDENT               	i
=                   	=
INT                 	4
NEW_LINE            	

CALL                	call
IDENT               	square_cube
(                   	(
IDENT               	i
,                   	,
IDENT               	isq
,                   	,
IDENT               	icub
)                   	)
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
STRING              	"i,i^2,i^3="
,                   	,
IDENT               	i
,                   	,
IDENT               	isq
,                   	,
IDENT               	icub
NEW_LINE            	

IDENT               	print
*                   	*
,                   	,
STRING              	"i,i^2,i^3="
,                   	,
IDENT               	i
,                   	,
IDENT               	isq
,                   	,
IDENT               	icub
NEW_LINE            	

END                 	end
PROGRAM             	program
IDENT               	xx
NEW_LINE            	

